malariagen_data/anoph/cnv_data.py

```diff
@@ -80,8 +80,16 @@ def open_cnv_hmm(self, sample_set: base_params.sample_set) -> zarr.hierarchy.Gro
             release = self.lookup_release(sample_set=sample_set)
             release_path = self._release_to_path(release)
             path = f"{self._base_path}/{release_path}/cnv/{sample_set}/hmm/zarr"
-            store = init_zarr_store(fs=self._fs, path=path)
-            root = zarr.open_consolidated(store=store)
+
+            # If CNV HMM data exists for this sample set then return the zarr,
+            # Otherwise return None.
+            marker = path + "/.zmetadata"
+            if self._fs.exists(marker):
+                store = init_zarr_store(fs=self._fs, path=path)
+                root = zarr.open_consolidated(store=store)
+            else:
+                root = None
```

Comment:
```suggestion
            store = init_zarr_store(fs=self._fs, path=path)
            try:
                root = zarr.open_consolidated(store=store)
            except FileNotFoundError:
                root = None
```

Suggestion above follows approach taken in `open_haplotypes()`.

It's a bit better to ask forgiveness here rather than permission. The reason is that an `exists()` check will add an extra HTTP request which could take maybe 100ms, and these could add up if you have lots of sample sets to check.

Reply:
Thanks @alimanfoo , will do. I agree with the reason but not the maxim!